test/unit/attributes.js

```diff
@@ -1107,6 +1107,71 @@ QUnit.test( "val(select) after form.reset() (Bug #2551)", function( assert ) {
 	jQuery( "#kk" ).remove();
 } );
 
+QUnit.test( "select.val(space characters) (gh-2978)", function( assert ) {
```

Comment:
I guess I don't see the similarity in purpose between those `app-boot` tests and this. Looks like they have lots of helpers and they are establishing that HTML is generated as expected, or that certain callbacks are called/not called. We have some tests in ajax [that are similar](https://github.com/jquery/jquery/blob/master/test/unit/ajax.js#L40). I think it would help me if I could see how you'd like to write some of our existing tests to compare something closer to home.